### ISSUE TITLE
adding --backup-dir-deletes-only feature

### ISSUE THREAD

**Issue by cryptosmasher** [ncw/rclone#2352]
#### What is the problem you are having with rclone?
taking advantage of box versioning when using sync - see thread: 
https://forum.rclone.org/t/understanding-backup-dir-behavior-with-box-and-others/5880/7

#### What is your rclone version (eg output from `rclone -V`)
rclone v1.41
- os/arch: windows/amd64
- go version: go1.10

#### Which OS you are using and how many bits (eg Windows 7, 64 bit)
Windows Server 2012 R2 64 bit

####  Which cloud storage system are you using? (eg Google Drive)
box.com

#### The command you were trying to run (eg `rclone copy /tmp remote:tmp`)
clone sync

**Comment by ncw** [comment]
My understanding of the problem is that you'd like to use file versioning.

Without `--backup-dir` the box file versioning works perfectly but deleting files sends them into the trash which isn't well organised and doesn't have an SLA.

With `--backup-dir` the box file versioning stops working as rclone moved any updated files into the `--backup-dir` which means that versions are no longer kept in the destination.  However `--backup-dir` works well for capturing deleted files in a more organised way than putting them in the trash.

So an option which only put deleted files, not overwritten files into the `--backup-dir` would solve the problem.

**Comment by ncw** [comment]
@cryptosmasher do you want to have a go at this?  I can talk you through what needs to be done?

**Comment by cryptosmasher** [comment]
Yes that sounds ideal for box.com - I have been using rclone lately on windows and don’t particularly have an environment to compile but I am happy to help and get that going

**Comment by cryptosmasher** [comment]
I'm afraid I took a quickstab at sync.go - considering I don't know go at all I'm a bit lost - and for that matter I have to read up on how to properly compile go applications on windows

**Comment by ncw** [comment]
@cryptosmasher sorry for the delayed reply!  Do you want to carry on or do you want me to find another ~victim~ volunteer?

**Comment by ncw** [comment]
@cryptosmasher no worries - thanks for having a go.

**Comment by cfbao** [comment]
I'd like to comment that this issue is not specific to Box. This feature can benefit many other remotes.

@ncw Do you think this is a good "first issue" for someone new? I might take a stab at it.

**Comment by ncw** [comment]
> I'd like to comment that this issue is not specific to Box. This feature can benefit many other remotes.

Indeed.

> @ncw Do you think this is a good "first issue" for someone new? I might take a stab at it. (no promise though...)

Yes I think it would be a great first issue.

**Comment by wes1993** [comment]
Hello, I really need this feature, there are some news?

**Comment by garrettisrael** [comment]
Any update on this @ncw? I would really like to implement this feature into my scripts. Thanks.

**Comment by lntutor** [comment]
I am working on this. I plan to add the requested --backup-dir-deletes-only modifier, with sync tests covering both overwritten and deleted destination files, plus documentation.
